internal/infrastructure/kubernetes/infra_resource.go

```diff
@@ -159,6 +188,18 @@ func (i *Infra) deleteDeployment(ctx context.Context, r ResourceRender) error {
 	return i.Client.Delete(ctx, deployment)
 }
 
+// deleteDaemonSet deletes the Envoy DaemonSet in the kube api server, if it exists.
+func (i *Infra) deleteDaemonSet(ctx context.Context, r ResourceRender) error {
+	daemonSet := &appsv1.DaemonSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: i.Namespace,
+			Name:      r.Name(),
+		},
+	}
+
+	return i.Client.Delete(ctx, daemonSet)
```

Comment:
Maybe worth ignoring `NotFound `errors in the deletion flow.

Reply:
[Alredy exist](https://github.com/zvlb/gateway/blob/ds/internal/infrastructure/kubernetes/infra_client.go#L54)